lib/platform.cpp

```diff
@@ -152,9 +152,17 @@ bool Platform::set(Type t)
 
 bool Platform::set(const std::string& platformstr, std::string& errstr, const std::vector<std::string>& paths, bool verbose)
 {
-    if (platformstr == "win32A")
+    if (platformstr == "win32A") {
+        std::cout << "Platform 'win32A' is deprecated and will be removed in a future version. Please use 'win32a' instead." << std::endl;
         set(Type::Win32A);
-    else if (platformstr == "win32W")
+    }
+    else if (platformstr == "win32a")
+        set(Type::Win32A);
+    else if (platformstr == "win32W") {
+        std::cout << "Platform 'win32W' is deprecated and will be removed in a future version. Please use 'win32w' instead." << std::endl;
```

Comment:
this message will not be handled well by some plugins. i.e. the vscode plugin I use will determine that cppcheck output is invalid and will not show any warnings..
I think we usually write something like "cppcheck: error: ..." when there are configuration problems.
however I fear that even "cppcheck: error: ..." would not be handled well by all plugins ideally we would report all the issues using proper ErrorMessage output

Reply:
If that would cause plugins to break we already did that quite a while ago with `checkersReport` (that actually broke the CLion plugin) and the "active checks" output as well as other deprecations way before that.

> but I think writing proper ErrorMessage output can be done in a separate future PR.

Yes, so that everything results in a XML entry. I have that planned. After I am done with all my current tasks that will be the next to look into as it blocks further deprecations and changes like this.

Reply:
> If that would cause plugins to break we already did that quite a while ago with checkersReport

it is reported as a normal information message nowadays.

Reply:
I think that we could at least always use the same format for bad command line arguments:
```
$ ./cppcheck --not-exist
cppcheck: error: unrecognized command line option: "--not-exist".
```

Reply:
> I think that we could at least always use the same format for bad command line arguments:

That is (implicitly) part of the plan.

Reply:
I will introduce the lower-case aliases without a deprecation warning.

Reply:
I filed https://trac.cppcheck.net/ticket/13262 about deprecating the mixed case versions.